### PR TITLE
Fix global index destruction [HZ-3218] 

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast-enterprise/issues/6507

EE: https://github.com/hazelcast/hazelcast-enterprise/pull/6684

Destruction of global indexes must be done one time and at the end of the destroy flow. Currently it is destroyed inside partitions and if a later partition wants to clear data from global index it couldn't find it alive(since it has been disposed early)

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-ee-tests

File: hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java [hunk 1]
```diff
@@ -382,29 +381,6 @@ public ConcurrentMap<Integer, IndexRegistry> getPartitionedIndexRegistry() {
...
-    }
-
-    private void add(Consumer<IndexConfig> consumer, IndexRegistry indexRegistry) {
-        InternalIndex[] indexes = indexRegistry.getIndexes();
-        for (int i = 0; i < indexes.length; i++) {
-            IndexConfig config = indexes[i].getConfig();
-            consumer.accept(config);
-        }
-    }
-
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java:407]: Based on what I see in my local, this part is not related to the fix, just removing dead code.

**Review by blazember — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java:407]: true, it was dead code.

**Comment by ahmetmircik** [comment]
run-ee-tests

**Review by vbekiaris — COMMENTED** [review]
LGTM but there seems to be an issue with TS global index destruction in EE tests

**Comment by ahmetmircik** [comment]
@vbekiaris thx, already fixed it here --> https://github.com/hazelcast/hazelcast/pull/25742
